src/Core/Config/PreloadPConfigAdapter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+use dba;
+use Exception;
+use Friendica\BaseObject;
+
+require_once 'include/dba.php';
+
+/**
+ * Preload PConfigAdapter
+ *
+ * Minimize the number of database queries to retrieve configuration values at the cost of memory.
+ *
+ * @author Hypolite Petovan <mrpetovan@gmail.com>
+ */
+class PreloadPConfigAdapter extends BaseObject implements IPConfigAdapter
+{
+	private $config_loaded = false;
+
+	public function __construct($uid)
+	{
+		$this->load($uid, 'config');
+	}
+
+	public function load($uid, $family)
+	{
+		if ($this->config_loaded) {
+			return;
+		}
+
+		$a = self::getApp();
+
+		$pconfigs = dba::select('pconfig', ['cat', 'v', 'k'], ['uid' => $uid]);
+		while ($pconfig = dba::fetch($pconfigs)) {
+			$cat   = $pconfig['cat'];
+			$k     = $pconfig['k'];
+			$value = (preg_match("|^a:[0-9]+:{.*}$|s", $pconfig['v']) ? unserialize($pconfig['v']) : $pconfig['v']);
+
+			$a->config[$uid][$cat][$k] = $value;
+		}
+		dba::close($pconfigs);
+
+		$this->config_loaded = true;
+	}
+
+	public function get($uid, $cat, $k, $default_value = null, $refresh = false)
+	{

```

Comment:
See my comment above. We have to be able to fetch completely new content.

Reply:
Done.